state-chain/pallets/cf-pools/src/migrations/v1.rs

```diff
@@ -21,21 +26,25 @@ mod old {
 		pub pool_state: cf_amm::v1::PoolState<(T::AccountId, OrderId)>,
 	}
 
-	#[cfg(feature = "try-runtime")]
 	#[frame_support::storage_alias]
 	pub type Pools<T: Config> =
 		StorageMap<Pallet<T>, Twox64Concat, CanonicalAssetPair, Pool<T>, OptionQuery>;
 }
 
 impl<T: Config> OnRuntimeUpgrade for Migration<T> {
 	fn on_runtime_upgrade() -> frame_support::weights::Weight {
-		Pools::<T>::translate::<old::Pool<T>, _>(|_key, old_pool: old::Pool<T>| {
-			Some(Pool::<T> {
-				range_orders_cache: old_pool.range_orders_cache,
-				limit_orders_cache: old_pool.limit_orders_cache,
-				pool_state: old_pool.pool_state.into(),
-			})
-		});
+		for (key, pool) in old::Pools::<T>::iter().filter_map(|(old_key, old_pool)| {
```

Comment:
I think this should use `drain` otherwise it won't remove the old pools (they will remain stored under the old key).